Vacuum/Clean.fs

```diff
@@ -173,14 +178,24 @@ let clean({
     info $"Cleaning directory %s{directory.RawPathString}"
     if bytesToFree.IsSome then
         info $"Cleaning %d{bytesToFree.Value} bytes"
+    elif freeUntil.IsSome then
+        info $"Cleaning until %d{freeUntil.Value} bytes are free"
 
     let itemsBefore = itemCount directory
     let diskRoot = Path.GetPathRoot directory.RawPathString
     let freeDiskSpaceBefore = NativeFunctions.getFreeDiskSpace diskRoot
 
+    let bytesToFree' =
+        match bytesToFree, freeUntil with
+        | Some _, None -> bytesToFree
+        | None, Some bytes ->
+            Some <| bytes - freeDiskSpaceBefore
```

Comment:
One minor issue here: this number may me less than zero, and we don't want that. Let's add `max 0 (bytes - freeDiskSpaceBefore)`?

Reply:
I don't think this will cause issues because `takeBytes` will filter it out.
After all, someone might also try passing a negative number on the command line and there aren't any checks for that either.

Reply:
```f#
let private takeBytes bytes (files: AbsolutePath seq): IFileSystemItem seq =
    seq {
        use enumerator = files.GetEnumerator()
        let mutable currentSize = 0L
        while currentSize < bytes && enumerator.MoveNext() do // This loop will immediately terminate if bytes is less than zero
            let entry = enumerator.Current
            let entrySize = getEntrySize entry

            currentSize <- currentSize + entrySize
            yield SizedFileSystemItem(entry, entrySize)
    }
```

Reply:
My concern is that it will report a negative amount of bytes to be cleaned in the command line. I expect it to be zero in this case :)

Reply:
Yeah, that makes sense. I'll add it :)